packages/next-app/src/contexts/PoolMetadataContext.tsx

```diff
@@ -88,6 +93,16 @@ export function PoolMetadataProvider({ children }: PropsWithChildren) {
     return metadata.every((item) => toSlug(item.key) !== toSlug(key));
   }
 
+  function handleGoToPool(formData: PoolAttribute) {
+    const { push } = useRouter();
+    const { chain } = useNetwork();
+
+    const networkId = formData.network ?? chain?.id.toString();
+    const networkName =
+      networkIdEnumMap[networkId as keyof typeof networkIdEnumMap];
+    push(`/metadata/${networkName}/pool/${formData.poolId}`);
+  }
+
```

Comment:
Not sure I follow why define this function within the context -- are we using anything that's context dependent here?

Reply:
It is not, but it was a way to share the function between the layout (sidebar) and the pages. I could also put it inside a handler folder. WDYT?

Reply:
I'd suggest only keep things in the context that need to be there and if want to share that in two files just separate it into another file